manager/src/lib.rs

```diff
@@ -150,8 +138,27 @@ impl CronManager {
             available_balance: 0,
             staked_balance: 0,
             agent_fee: u128::from(GAS_BASE_FEE),
-            slot_granularity: SLOT_GRANULARITY
-        }
+            slot_granularity: SLOT_GRANULARITY,
+            agent_storage_usage: 0
+        };
+        this.measure_account_storage_usage();
+        this
+    }
+
+    /// Measure the storage an agent will take and need to provide
+    fn measure_account_storage_usage(&mut self) {
+        let initial_storage_usage = env::storage_usage();
+        // Create a temporary, dummy entry and measure the storage used.
+        let tmp_account_id = "a".repeat(64);
+        let tmp_agent = Agent {
+            payable_account_id: tmp_account_id.clone(),
+            balance: U128::from(0),
+            total_tasks_executed: U128::from(0)
+        };
+        self.agents.insert(&tmp_account_id, &tmp_agent);
+        self.agent_storage_usage = env::storage_usage() - initial_storage_usage;
+        // Remove the temporary entry.
+        self.agents.remove(&tmp_account_id);
```

Comment:
Interesting -- guess there's no in-memory way huh?

Reply:
not that I'm aware of, this is how Eugene has been doing it in our smart contracts